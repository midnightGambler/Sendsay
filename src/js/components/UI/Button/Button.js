import React from "react";

const Button = ({ className = '', children, ...rest }) => (
  <button className={`btn ${className}`} {...rest}>
    {children}
  </button>
);

export default Button;