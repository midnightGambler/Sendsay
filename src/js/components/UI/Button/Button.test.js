import React from "react";
import Button from "./Button";
import { shallow } from "enzyme";

it("renders Button correctly", () => {
  const btn = shallow(<Button>Кнопушка</Button>);
  expect(btn).toMatchSnapshot();
});

it("test button onClick event", () => {
  const mockedCallback = jest.fn();
  const btn = shallow(<Button onClick={mockedCallback}>Кнопушка</Button>);
  btn.find("button").simulate("click");
  expect(mockedCallback.mock.calls.length).toBe(1);
});
