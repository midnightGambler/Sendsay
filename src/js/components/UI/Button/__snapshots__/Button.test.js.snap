// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Button correctly 1`] = `
<button
  className="btn "
>
  Кнопушка
</button>
`;
